### ISSUE TITLE
How to recall UIViewController without reload data

### ISSUE THREAD

**Issue by hprobotic** [John-Lluch/SWRevealViewController#609]
Hi

I have 3 view already open with name: View A, View B and View C
For example: User working on View A, after done, move to view B for completed some step but if i use revealViewController for push View B from current view (View A), View B alway be reload and break user workflow

**Comment by iDevelopper** [comment]
Why do not push view B with a navigation controller, if one exist, or presentViewController?

**Comment by hprobotic** [comment]
Can i presentViewController B without losing navigation bar in View B?

I tired it before but it will just load Single View B without load SW Navigation bar

**Comment by iDevelopper** [comment]
If view controller A is in a navigation controller push the view controller B with it and you will have a back button to come back.

**Comment by hprobotic** [comment]
Ahh, we have some confused here

As i has been explain:
I have 3 view controller A (Create Poll), B (List Poll), C.

When i press Done Button on View A (Creat Poll), i want present View B (List Poll) immediately.

Here is what i did

```
func onDonePressed(sender: UIButton!) {
            let votePollViewController = PollListViewController(queryType: PollListQuery.All)
            let navigationViewController = UINavigationController(rootViewController: votePollViewController)
            self.revealViewController().pushFrontViewController(navigationViewController, animated: true)
}
```

But this code will be reload view B like when we press button to call view B on Side Menu

**Comment by iDevelopper** [comment]
Yes you replace the front view controller A by a new B

**Comment by hprobotic** [comment]
Hi
Thanks you. Can you give me an example?

**Comment by iDevelopper** [comment]
Yes if you tell me what you want to do exactly! If you want recall B without reload datas, you have to save it in a variable and not allocate it each time.

**Comment by hprobotic** [comment]
Hi
Here is GIF version: http://imgur.com/CiI7UZE
I want present View B (List Poll) without reload data for collection view when i press Done in View A (Create Poll)
Currently, whenever i call View B, it will reload all data from Server

**Comment by iDevelopper** [comment]
If you want recall B without reload datas, you have to save it in a variable the first time you create it and not allocate it each time.

let votePollViewController = PollListViewController(queryType: PollListQuery.All) create a new instance of the controller!

**Comment by iDevelopper** [comment]
Have a look at this sample:

You will see that mapController1 and mapController2 are saved in two variables in MenuTableViewController, and in didSelectRow, i test if they are already existing.

[MapSample.zip](https://github.com/John-Lluch/SWRevealViewController/files/380199/MapSample.zip)

**Comment by hprobotic** [comment]
Wow, looking on your example, it work like charm, thanks

**Comment by iDevelopper** [comment]
Welcome!

If you are interested by menus on top I wrote another library inspired from SW (SW is no longer maintained I think, as John does not respond to the issues anymore).

When I saw your gif, I thought this could be a good test!

https://github.com/iDevelopper/PBRevealViewController

**Comment by hprobotic** [comment]
Ok, sure
Just take a look and started your project few minutes ago. Can you give me your facebook id or skype for further discuss 💃

**Comment by iDevelopper** [comment]
I would prefer mails as I don't use FB a lot!

patrick.bodet4@wanadoo.fr
